js/render/draw_extrusion.js

```diff
@@ -0,0 +1,292 @@
+'use strict';
+
+var mat3 = require('gl-matrix').mat3;
+var mat4 = require('gl-matrix').mat4;
+var vec3 = require('gl-matrix').vec3;
+var pixelsToTileUnits = require('../source/pixels_to_tile_units');
+var Buffer = require('../data/buffer');
+var VertexArrayObject = require('./vertex_array_object');
+var StructArrayType = require('../util/struct_array');
+
+module.exports = draw;
+
+function draw(painter, source, layer, coords) {
+    if (layer.paint['fill-opacity'] === 0) return;
+    var gl = painter.gl;
+    gl.disable(gl.STENCIL_TEST);
+    painter.depthMask(true);
+
+    // Create a new texture to which to render the extrusion layer. This approach
+    // allows us to adjust opacity on a per-layer basis (eliminating the interior
+    // walls per-feature opacity problem)
+    var texture = new ExtrusionTexture(gl, painter, layer);
+    texture.bindFramebuffer();
+
+    gl.clearStencil(0x80);
+    gl.stencilMask(0xFF);
+    gl.clear(gl.COLOR_BUFFER_BIT | gl.DEPTH_BUFFER_BIT);
+    gl.stencilMask(0x00);
```

Comment:
Here's what I've figured out:
* `gl.clear` only affects the currently bound framebuffer and is needed here
* I can delete several of these lines (basically all but the `gl.clear`) and get visually the same results. My best guess is that this was in here because the original implementation (from which I copied these) rendered to the map as a normal layer, so it needed to fiddle with the stencil mask (so as to clear the entire screen). Now that we're rendering in a new framebuffer, it doesn't need to redraw a mask that's not actually there — so, deleting all these lines except the `gl.clear`, which clears its last frame draw. (Fwiw/for future reference: based on my testing I actually don't think it was affecting the global stencil mask anyway because of the different framebuffer.)
* testing for the clipping masks made me realize I forgot to flip the depthMask back off after this draw, which was causing fills to z-fight if above an extrusion layer — fixed 😅